src/rviz_visual_tools.cpp

```diff
@@ -971,14 +971,18 @@ bool RvizVisualTools::publishCone(const geometry_msgs::Pose& pose, double angle,
 bool RvizVisualTools::publishABCDPlane(const double A, const double B, const double C, const double D,
                                         colors color, double x_width, double y_width)
 {
-  Eigen::Isometry3d pose;
+  if (D > 0)
+  {
+    ROS_WARN_STREAM_NAMED(LOGNAME, "For a plane equation Ax+By+Cz+D=0, D should be <=0.");
+  }
 
   // The coefficients A,B,C give the normal to the plane.
   Eigen::Vector3d n(A, B, C);
 
   // Graphic is centered at this point
-  Eigen::Vector3d center = n.normalized() * D;
+  Eigen::Vector3d center = n.normalized() * sqrt(-D);
```

Comment:
what's the purpose of this?

Reply:
See above comment. I could make it an absolute value if you want.

Reply:
I guess it probably should be abs(), not negative

Reply:
I agree that I made a small mistake here
```suggestion
  Eigen::Vector3d center = n * D / n.norm();
```

Reply:
You just need to adapt the sign of D depending on what plane equation you use.

Reply:
What works well is either:

```
  double distance = D / n.norm();
  Eigen::Vector3d center = - distance * n.normalized();
```

or

`Eigen::Vector3d center = n.normalized() * sqrt(fabs(D));`

I'll keep the first b/c it's more readable. Can send screenshots if you want confirmation

Reply:
> What works well is either:
> 
> ```
>   double distance = D / n.norm();
>   Eigen::Vector3d center = - distance * n.normalized();
> ```

This is exactly the version I suggested. Just make sure you use the correct sign.


> or
> 
> `Eigen::Vector3d center = n.normalized() * sqrt(fabs(D));`
> 

This version is problematic since it doesn't normalize D properly and doesn't account for negative direction!

Reply:
A test of negative x/y:

![negative_values_n normalized() * sqrt(fabs(D))](https://user-images.githubusercontent.com/11284393/58199135-72d45480-7c95-11e9-9acd-dbcaac1deb4e.png)

A test of positive x/y:

![n normalized() * sqrt(fabs(D))](https://user-images.githubusercontent.com/11284393/58199161-7ff14380-7c95-11e9-9dae-f88639b7d6e7.png)

Reply:
what exactly are your ABCD values?

Reply:
I think the way you generate the planes is flawed. A,B can't be used to determine the position of the plane, so you should rather generate your normals from 0° to 90° and then compute the center point from you desired target position. ABCD does not allow "moving" the plane to a desired position so if you really want to visualize definite cubes you should add some kind of target point or x/y offset.

Reply:
Also the results you printed look correct considering the planes all point to the center

Reply:
> This is exactly the version I suggested

Yep, you mostly called that. Good job.

> what exactly are your ABCD values?

They're in rviz_visual_tools_demo.cpp. A = [0:0.1:1], B = 2.6, C = 0, D varies to satisfy Ax+By+Cz+D=0.

> ABCD does not allow "moving" the plane to a desired position so if you really want to visualize definite cubes you should add some kind of target point or x/y offset or even better a reference pose.

We're going in circles now. Remember, the original function was like that: 

`publishABCDPlane(A, B, C, std::vector<double>& center, ...);`

But you requested to add a D argument. I think it's better as it is currently -- it sounds like most people want to use this to publish a plane from PCL, which provides ABCD parameters. We could add an overloaded function that is like the original version (probably in another PR). I would probably call it `publishABCCenterPlane()`.

Reply:
> We're going in circles now. Remember, the original function was like that:

> `publishABCDPlane(A, B, C, std::vector<double>& center, ...);`

The original function was not ABCD (as requested in the issue) but some variant of the normal form. So either D had to be added or the function name should be changed. 

> We could add another function that is like the original version (probably in another PR). I would probably call it publishABCCenterPlane()

I think this use case is fully covered by `publishCuboid(const Eigen::Isometry3d& pose, double depth, double width, double height, colors color)` since the only advantage would be to compute the rotation from a normal vector (which is a one-liner). Of course you could just add a wrapper function that does that.